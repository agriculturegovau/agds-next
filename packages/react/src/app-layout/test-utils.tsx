import { Fragment } from 'react';
import { Text } from '../text';
import { NotificationBadge } from '../notification-badge';
import { VisuallyHidden } from '../a11y';
import {
	ChartLineIcon,
	EmailIcon,
	ExitIcon,
	HelpIcon,
	HomeIcon,
	SuccessIcon,
	SettingsIcon,
	FactoryIcon,
	AvatarIcon,
	ChevronsLeftIcon,
	MinusIcon,
} from '../icon';
import {
	DropdownMenuDivider,
	DropdownMenuItem,
	DropdownMenuItemLink,
	DropdownMenuPanel,
	DropdownMenuGroup,
	DropdownMenuGroupLink,
	DropdownMenuItemRadio,
} from '../dropdown-menu';

export const navigationItems = (businessName: string) => [
	{
		options: { disableGroupPadding: true },
		items: [
			{
				label: 'Back to my account',
				icon: ChevronsLeftIcon,
				href: '/account',
			},
		],
	},
	{
		options: { disableGroupPadding: true },
		items: [
			{
				label: (
					<Fragment>
						<Text fontWeight="bold" fontSize="xs">
							{businessName}
						</Text>
						<Text color="muted" fontSize="xs">
							ABN: 00 000 000 000
						</Text>
					</Fragment>
				),
			},
		],
	},
	[
		{
			label: 'Dashboard',
			icon: HomeIcon,
			href: '/',
		},
		{
			label: 'Establishments',
			icon: FactoryIcon,
			href: '/establishments',
			items: [
				{
					label: 'Canberra',
					icon: MinusIcon,
					href: '/establishments/canberra',
				},
				{
					label: 'Sydney',
					icon: MinusIcon,
					href: '/establishments/sydney',
				},
			],
		},
		{
			label: 'Data and Insights',
			icon: ChartLineIcon,
			href: '/intelligence',
		},
		{
			label: 'Compliance',
			icon: SuccessIcon,
			href: '/compliance',
			items: [
				{
					label: 'Audit',
					icon: MinusIcon,
					href: '/compliance/audit',
				},
				{
					label: 'Certificates',
					icon: MinusIcon,
					href: '/compliance/certificates',
				},
			],
		},
	],

	[
		{
			label: 'Help',
			icon: HelpIcon,
			href: '/help',
		},
	],
	[
		{
			label: 'Sign out',
			onClick: console.log,
			icon: ExitIcon,
		},
	],
];

export const exampleData = {
	userNames: {
		short: 'Ab Cd',
		regular: 'Toto Wolff',
		medium: 'Benjamin Harrington',
		long: 'Benjamin Alexander Harrington-Smythe',
	},
	businessNames: {
		short: ['Lorem', 'Ipsum', 'Company'],
		regular: ['Antfix', 'Produce Fresh', 'Organic Co'],
		medium: [
			'Malesuada lacus a tortor blandit fermentum phasellus',
			'Iaculis tortor duis ante nec risus elementum id ui',
			'Purus tortor lacus malesuada phasellus ipsum ex duis libero ante id',
		],
		long: [
			'Duis quis iaculis libero nec consectetur luctus ex in malesuada lacus a tortor blandit fermentum phasellus iaculis ipsum nec purus aliquet id elementum risus duis ante ui',
			'Duis ut a libero duis nisl ex lacus in luctus viverra iaculis iaculis tortor duis malesuada blandit purus diam ipsum ante nec risus elementum id ui nec quis aliquet phasellus fermentum',
			'Blandit iaculis iaculis quis ante diam viverra elementum ui risus nec luctus purus tortor lacus malesuada phasellus ipsum ex duis libero ante id Duis fermentum nisl aliquet duis in a nec ut consectetur',
		],
	},
};

export function ExampleAccountDropdown({
	businesses = exampleData.businessNames.regular,
	onBusinessChange,
	selectedBusinessName,
}: {
	businesses?: string[];
	onBusinessChange: (value: string) => void;
	selectedBusinessName: string;
}) {
	return (
		<DropdownMenuPanel palette="light">
			<DropdownMenuGroup label="Businesses">
				{businesses.map((businessName) => (
					<DropdownMenuItemRadio
						key={businessName}
						checked={businessName === selectedBusinessName}
						secondaryText="ABN 00 000 000 000"
						onClick={() => onBusinessChange?.(businessName)}
					>
						{businessName}
					</DropdownMenuItemRadio>
				))}
				<DropdownMenuGroupLink href="#">View all</DropdownMenuGroupLink>
			</DropdownMenuGroup>
			<DropdownMenuDivider />
			<DropdownMenuGroup label="My account">
				<DropdownMenuItemLink href="/profile" icon={AvatarIcon}>
					Profile
				</DropdownMenuItemLink>
				<DropdownMenuItemLink
					href="/messages"
					icon={EmailIcon}
					endElement={
						<span>
							<NotificationBadge tone="action" value={6} max={99} aria-hidden />
							<VisuallyHidden>, 6 unread</VisuallyHidden>
						</span>
					}
				>
					Messages
				</DropdownMenuItemLink>
				<DropdownMenuItemLink href="/account-settings" icon={SettingsIcon}>
					Account settings
				</DropdownMenuItemLink>
			</DropdownMenuGroup>
			<DropdownMenuDivider />
			<DropdownMenuItem onClick={() => console.log('sign out')} icon={ExitIcon}>
				Sign out
			</DropdownMenuItem>
		</DropdownMenuPanel>
	);
}
