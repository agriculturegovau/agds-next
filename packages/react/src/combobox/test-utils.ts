export type Option = { label: string; value: string };

export const COUNTRY_OPTIONS = [
	'Afghanistan is a really long country that spans multiple lines cmon this can be way longer than that Afghanistan is a really long country that spans multiple lines cmon this can be way longer than that',
	'Albania',
	'Algeria',
	'Andorra',
	'Angola',
	'Antigua and Barbuda',
	'Argentina',
	'Armenia',
	'Australia',
	'Austria',
	'Azerbaijan',
	'Bahamas',
	'Bahrain',
	'Bangladesh',
	'Barbados',
	'Belarus',
	'Belgium',
	'Belize',
	'Benin',
	'Bhutan',
	'Bolivia',
	'Bosnia and Herzegovina',
	'Botswana',
	'Brazil',
	'Brunei',
	'Bulgaria',
	'Burkina Faso',
	'Burundi',
	"CÃ´te d'Ivoire",
	'Cabo Verde',
	'Cambodia',
	'Cameroon',
	'Canada',
	'Central African Republic',
	'Chad',
	'Chile',
	'China',
	'Colombia',
	'Comoros',
	'Congo (Congo-Brazzaville)',
	'Costa Rica',
	'Croatia',
	'Cuba',
	'Cyprus',
	'Czechia (Czech Republic)',
	'Democratic Republic of the Congo',
	'Denmark',
	'Djibouti',
	'Dominica',
	'Dominican Republic',
	'Ecuador',
	'Egypt',
	'El Salvador',
	'Equatorial Guinea',
	'Eritrea',
	'Estonia',
	'Eswatini (fmr. "Swaziland")',
	'Ethiopia',
	'Fiji',
	'Finland',
	'France',
	'Gabon',
	'Gambia',
	'Georgia',
	'Germany',
	'Ghana',
	'Greece',
	'Grenada',
	'Guatemala',
	'Guinea',
	'Guinea-Bissau',
	'Guyana',
	'Haiti',
	'Holy See',
	'Honduras',
	'Hungary',
	'Iceland',
	'India',
	'Indonesia',
	'Iran',
	'Iraq',
	'Ireland',
	'Israel',
	'Italy',
	'Jamaica',
	'Japan',
	'Jordan',
	'Kazakhstan',
	'Kenya',
	'Kiribati',
	'Kuwait',
	'Kyrgyzstan',
	'Laos',
	'Latvia',
	'Lebanon',
	'Lesotho',
	'Liberia',
	'Libya',
	'Liechtenstein',
	'Lithuania',
	'Luxembourg',
	'Madagascar',
	'Malawi',
	'Malaysia',
	'Maldives',
	'Mali',
	'Malta',
	'Marshall Islands',
	'Mauritania',
	'Mauritius',
	'Mexico',
	'Micronesia',
	'Moldova',
	'Monaco',
	'Mongolia',
	'Montenegro',
	'Morocco',
	'Mozambique',
	'Myanmar (formerly Burma)',
	'Namibia',
	'Nauru',
	'Nepal',
	'Netherlands',
	'New Zealand',
	'Nicaragua',
	'Niger',
	'Nigeria',
	'North Korea',
	'North Macedonia',
	'Norway',
	'Oman',
	'Pakistan',
	'Palau',
	'Palestine State',
	'Panama',
	'Papua New Guinea',
	'Paraguay',
	'Peru',
	'Philippines',
	'Poland',
	'Portugal',
	'Qatar',
	'Romania',
	'Russia',
	'Rwanda',
	'Saint Kitts and Nevis',
	'Saint Lucia',
	'Saint Vincent and the Grenadines',
	'Samoa',
	'San Marino',
	'Sao Tome and Principe',
	'Saudi Arabia',
	'Senegal',
	'Serbia',
	'Seychelles',
	'Sierra Leone',
	'Singapore',
	'Slovakia',
	'Slovenia',
	'Solomon Islands',
	'Somalia',
	'South Africa',
	'South Korea',
	'South Sudan',
	'Spain',
	'Sri Lanka',
	'Sudan',
	'Suriname',
	'Sweden',
	'Switzerland',
	'Syria',
	'Tajikistan',
	'Tanzania',
	'Thailand',
	'Timor-Leste',
	'Togo',
	'Tonga',
	'Trinidad and Tobago',
	'Tunisia',
	'Turkey',
	'Turkmenistan',
	'Tuvalu',
	'Uganda',
	'Ukraine',
	'United Arab Emirates',
	'United Kingdom',
	'United States of America',
	'Uruguay',
	'Uzbekistan',
	'Vanuatu',
	'Venezuela',
	'Vietnam',
	'Yemen',
	'Zambia',
	'Zimbabwe',
].map((country) => ({ label: country, value: country }));

export const STATE_OPTIONS = [
	{ label: 'Australian Capital Territory', value: 'act' },
	{ label: 'New South Wales', value: 'nsw' },
	{ label: 'Northern Territory', value: 'nt' },
	{ label: 'Queensland', value: 'qld' },
	{ label: 'South Australia', value: 'sa' },
	{ label: 'Tasmania', value: 'tas' },
	{ label: 'Victoria', value: 'vic' },
	{ label: 'Western Australia', value: 'wa' },
];

export const NAME_OPTIONS = [
	{
		fullName: 'John Doe',
		jobTitle: 'Product Manager',
		status: 'Draft',
		unreadMessageCount: 0,
	},
	{
		fullName: 'Jane Smith',
		jobTitle: 'Software Engineer',
		status: 'Approved',
		unreadMessageCount: 4,
	},
	{
		fullName: 'Robert Johnson',
		jobTitle: 'UX Designer',
		status: 'Deactivated',
		unreadMessageCount: 5,
	},
	{
		fullName: 'Emily Davis',
		jobTitle: 'Product Manager',
		status: 'Approved',
		unreadMessageCount: 6,
	},
	{
		fullName: 'Michael Williams',
		jobTitle: 'Software Engineer',
		status: 'Approved',
		unreadMessageCount: 0,
	},
	{
		fullName: 'Sarah Wilson',
		jobTitle: 'UX Designer',
		status: 'Approved',
		unreadMessageCount: 0,
	},
	{
		fullName: 'David Brown',
		jobTitle: 'Product Manager',
		status: 'Deactivated',
		unreadMessageCount: 0,
	},
	{
		fullName: 'Olivia Taylor',
		jobTitle: 'Software Engineer',
		status: 'Approved',
		unreadMessageCount: 6,
	},
	{
		fullName: 'James Moore',
		jobTitle: 'UX Designer',
		status: 'Approved',
		unreadMessageCount: 0,
	},
	{
		fullName: 'Sophia Clark',
		jobTitle: 'Product Manager',
		status: 'Approved',
		unreadMessageCount: 0,
	},
].map((option) => ({
	...option,
	label: option.fullName,
	value: option.fullName,
}));
