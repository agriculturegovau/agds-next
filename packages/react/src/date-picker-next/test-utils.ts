import * as yup from 'yup';

export const errorMessage = 'Enter a valid date';

// `yup.date()` can sometimes give false positives with certain string values
// Fixes https://github.com/jquense/yup/issues/764
// Tests for this can be viewed in the tests for `DatePicker` and `DateRangePicker`
// IMPORTANT: This is for example purposes only. It should NOT be exported from the design system
export const yupDateField = yup
	.date()
	.transform((current, orig) => (typeof orig === 'string' ? orig : current))
	.typeError(errorMessage);
