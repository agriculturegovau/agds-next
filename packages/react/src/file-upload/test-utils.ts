import { FileStatus, FileWithStatus } from './utils';

export function createExampleFileWithStatus(status?: FileStatus) {
	const bits = ['example'];
	const file: FileWithStatus = new File(bits, 'example.txt', {
		type: 'text/plain',
	});
	file.status = status;
	return file;
}

export function createExampleImageFileWithStatus(status?: FileStatus) {
	const file: FileWithStatus = new File(['example'], 'example.jpg', {
		type: 'image/jpg',
	});
	file.status = status;
	return file;
}
