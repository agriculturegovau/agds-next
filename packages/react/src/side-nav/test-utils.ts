import { SideNavProps } from './SideNav';

export const defaultTestingProps: SideNavProps = {
	title: 'Lodging your tax return',
	titleLink: '/',
	collapseTitle: 'In this section',
	items: [
		{
			href: '/welcome',
			label: 'Welcome',
		},
		{
			href: '/do-you-need-to-lodge-a-tax-return',
			label: 'Do you need to lodge a tax return?',
		},
		{
			href: '/lodge-online',
			label: 'Lodge online',
			items: [
				{
					href: '/lodge-online/pre-filling',
					label: 'Pre-filling your online tax return',
				},
			],
		},
		{
			href: '/whats-new-for-individuals',
			label: "What's new for individuals",
		},
		{
			href: '/why-you-may-receive-a-tax-bill',
			label: 'Why you may receive a tax bill',
		},
		{
			href: '/in-detail',
			label: 'In detail',
			items: [
				{
					href: '/in-detail/record-keeping',
					label: 'Record keeping',
					items: [
						{
							href: '/in-detail/record-keeping/tax',
							label: 'Keeping your tax records',
						},
						{
							href: '/in-detail/record-keeping/incorrect-amounts',
							label: 'Incorrect amounts',
						},
					],
				},
				{
					href: '/tax-receipt',
					label: 'Tax receipt',
				},
				{
					href: '/pre-fill availability',
					label: 'Pre-fill availability',
				},
			],
		},
	],
	activePath: '/in-detail',
};
