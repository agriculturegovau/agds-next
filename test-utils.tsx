import { PropsWithChildren, ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Core } from '@ag.ds-next/react/core';
import { theme } from '@ag.ds-next/react/ag-branding';

function AllTheProviders({ children }: PropsWithChildren) {
	return <Core theme={theme}>{children}</Core>;
}

const customRender = (
	ui: ReactElement,
	options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
