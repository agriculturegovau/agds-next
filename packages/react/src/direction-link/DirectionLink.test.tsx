import '@testing-library/jest-dom';
import 'html-validate/jest';
import { axe, toHaveNoViolations } from 'jest-axe';
import { cleanup, render } from '../../../../test-utils';
import {
	DirectionLink,
	DirectionLinkProps,
	DirectionButton,
	DirectionButtonProps,
} from './DirectionLink';

expect.extend(toHaveNoViolations);

afterEach(cleanup);

const directions = [
	{
		direction: 'left',
		label: 'Back',
	},
	{
		direction: 'right',
		label: 'Next',
	},
	{
		direction: 'up',
		label: 'Top',
	},
	{
		direction: 'down',
		label: 'Skip to footer',
	},
] as const;

function renderDirectionLink(props: DirectionLinkProps) {
	return render(<DirectionLink {...props} />);
}

describe('DirectionLink', () => {
	for (const { direction, label } of directions) {
		describe(`direction: ${direction}`, () => {
			it('renders correctly', () => {
				const { container } = renderDirectionLink({
					direction,
					children: label,
				});
				expect(container).toMatchSnapshot();
			});

			it('renders valid HTML with no a11y violations', async () => {
				const { container } = renderDirectionLink({
					direction,
					children: label,
				});
				expect(container).toHTMLValidate({
					extends: ['html-validate:recommended'],
				});
				expect(await axe(container)).toHaveNoViolations();
			});
		});
	}
});

function renderDirectionButton(props: DirectionButtonProps) {
	return render(<DirectionButton {...props} />);
}

describe('DirectionButton', () => {
	for (const { direction, label } of directions) {
		describe(`direction: ${direction}`, () => {
			it('renders correctly', () => {
				const { container } = renderDirectionButton({
					direction,
					children: label,
				});
				expect(container).toMatchSnapshot();
			});

			it('renders valid HTML with no a11y violations', async () => {
				const { container } = renderDirectionButton({
					direction,
					children: label,
				});
				expect(container).toHTMLValidate({
					extends: ['html-validate:recommended'],
				});
				expect(await axe(container)).toHaveNoViolations();
			});
		});
	}
});
