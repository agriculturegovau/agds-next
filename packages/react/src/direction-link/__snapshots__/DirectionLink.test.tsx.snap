// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DirectionButton direction: down renders correctly 1`] = `
<div>
  <button
    class="css-1jfzxu5-BaseButton-boxStyles-BaseDirectionLink"
    type="button"
  >
    Skip to footer
    <svg
      aria-hidden="true"
      class="css-ub2kzo-Icon"
      clip-rule="evenodd"
      fill-rule="evenodd"
      focusable="false"
      height="24"
      role="img"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 3L12 20"
      />
      <path
        d="M17 16L12 21L7 16"
      />
    </svg>
  </button>
</div>
`;

exports[`DirectionButton direction: left renders correctly 1`] = `
<div>
  <button
    class="css-1jfzxu5-BaseButton-boxStyles-BaseDirectionLink"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="css-ub2kzo-Icon"
      clip-rule="evenodd"
      fill-rule="evenodd"
      focusable="false"
      height="24"
      role="img"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M21 12L4 12"
      />
      <path
        d="M8 17L3 12L8 7"
      />
    </svg>
    Back
  </button>
</div>
`;

exports[`DirectionButton direction: right renders correctly 1`] = `
<div>
  <button
    class="css-1jfzxu5-BaseButton-boxStyles-BaseDirectionLink"
    type="button"
  >
    Next
    <svg
      aria-hidden="true"
      class="css-ub2kzo-Icon"
      clip-rule="evenodd"
      fill-rule="evenodd"
      focusable="false"
      height="24"
      role="img"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3 12L20 12"
      />
      <path
        d="M16 7L21 12L16 17"
      />
    </svg>
  </button>
</div>
`;

exports[`DirectionButton direction: up renders correctly 1`] = `
<div>
  <button
    class="css-1jfzxu5-BaseButton-boxStyles-BaseDirectionLink"
    type="button"
  >
    Top
    <svg
      aria-hidden="true"
      class="css-ub2kzo-Icon"
      clip-rule="evenodd"
      fill-rule="evenodd"
      focusable="false"
      height="24"
      role="img"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 21V4"
      />
      <path
        d="M7 8L12 3L17 8"
      />
    </svg>
  </button>
</div>
`;

exports[`DirectionLink direction: down renders correctly 1`] = `
<div>
  <a
    class="css-5fo29w-TextLink-boxStyles-BaseDirectionLink"
  >
    Skip to footer
    <svg
      aria-hidden="true"
      class="css-ub2kzo-Icon"
      clip-rule="evenodd"
      fill-rule="evenodd"
      focusable="false"
      height="24"
      role="img"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 3L12 20"
      />
      <path
        d="M17 16L12 21L7 16"
      />
    </svg>
  </a>
</div>
`;

exports[`DirectionLink direction: left renders correctly 1`] = `
<div>
  <a
    class="css-5fo29w-TextLink-boxStyles-BaseDirectionLink"
  >
    <svg
      aria-hidden="true"
      class="css-ub2kzo-Icon"
      clip-rule="evenodd"
      fill-rule="evenodd"
      focusable="false"
      height="24"
      role="img"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M21 12L4 12"
      />
      <path
        d="M8 17L3 12L8 7"
      />
    </svg>
    Back
  </a>
</div>
`;

exports[`DirectionLink direction: right renders correctly 1`] = `
<div>
  <a
    class="css-5fo29w-TextLink-boxStyles-BaseDirectionLink"
  >
    Next
    <svg
      aria-hidden="true"
      class="css-ub2kzo-Icon"
      clip-rule="evenodd"
      fill-rule="evenodd"
      focusable="false"
      height="24"
      role="img"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3 12L20 12"
      />
      <path
        d="M16 7L21 12L16 17"
      />
    </svg>
  </a>
</div>
`;

exports[`DirectionLink direction: up renders correctly 1`] = `
<div>
  <a
    class="css-5fo29w-TextLink-boxStyles-BaseDirectionLink"
  >
    Top
    <svg
      aria-hidden="true"
      class="css-ub2kzo-Icon"
      clip-rule="evenodd"
      fill-rule="evenodd"
      focusable="false"
      height="24"
      role="img"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 21V4"
      />
      <path
        d="M7 8L12 3L17 8"
      />
    </svg>
  </a>
</div>
`;
